src/SuperSimpleTcp/SimpleTcpServer.cs

```diff
@@ -812,15 +815,19 @@ private async Task AcceptConnections()
             _isListening = false;
         }
 
-        private async Task<bool> StartTls(ClientMetadata client)
+        private async Task<bool> StartTls(ClientMetadata client, CancellationToken token)
         {
             try
             {
-                await client.SslStream.AuthenticateAsServerAsync(
-                    _sslCertificate,
-                    _settings.MutuallyAuthenticate,
-                    SslProtocols.Tls12,
-                    _settings.CheckCertificateRevocation).ConfigureAwait(false);
+                var sslServerAuthOptions = new SslServerAuthenticationOptions
```

Comment:
This class is not available in all frameworks that the library targets, integrating this PR but leaving this portion out.  Instead of using the options class and passing in a token, I'm opting to instead catch the TaskCanceledException or OperationCanceledException.